payu/models/model.py

```diff
@@ -198,6 +177,59 @@ def get_prior_restart_files(self):
             print("No prior restart files found: {error}".format(error=str(e)))
             return []
 
+    def expand_executable_path(self, exec):
+        """Given an executable, return the expanded executable path"""
+        # Check if exe is already an absolute path
+        if os.path.isabs(exec):
+            return exec
+
+        # Check if path set by loading user modules has been defined
+        module_added_path = self.expt.user_modules_path
+        if module_added_path is None:
+            print("payu: warning: Skipping searching for model executable " +
+                  "in $PATH set by user modules")
+            module_added_paths = []
+        elif module_added_path == '':
+            module_added_paths = []
+        else:
+            module_added_paths = module_added_path.split(':')
+
+        # Search for exe inside paths added to $PATH by user-defined modules
+        exec_paths = []
+        for path in module_added_paths:
+            exec_path = os.path.join(path, exec)
+            if os.path.exists(exec_path) and os.access(exec_path, os.X_OK):
+                exec_paths.append(exec_path)
+
+        if len(exec_paths) > 1:
+            raise ValueError(
+                f"Executable {exec} found in multiple $PATH paths added by " +
+                f"user-defined modules in `config.yaml`. Paths: {exec_paths}")
+        elif len(exec_paths) == 1:
+            return exec_paths[0]
+
+        # Else prepend the lab bin path to exec
+        return os.path.join(self.expt.lab.bin_path, exec)
+
+    def setup_executable_paths(self):
+        """Set model executable paths"""
+        self.exec_prefix = self.config.get('exe_prefix', '')
```

Comment:
I don't really understand `exec_prefix`. I don't recall it ever being used in a configuration, and it is just randomly added in front of the `exe` path in `experiment.run` so it feels orthogonal to this new functionality. Maybe I'm misunderstanding it's purpose.

Reply:
This is the commit that it was added in: https://github.com/payu-org/payu/commit/8df851bb845147f8ac35c0071a596774a63e0bd6. I left it in as I don't know if it is still useful for e.g. gdb, gprof and valgrind?

Reply:
I guess so. This is a pretty specialist requirement, so we'll assume whomever is using it knows what they're doing and we don't have to hold their hand, so leave as-is.